File: 110.high_level_syntax/src/main/java/javax/HighLevelSyntax.java
package javax;

public class HighLevelSyntax {
    private static String buildString(final String one, final String two) {
        return one + " " + two;
    }

    private static String getParameter(String[] args) {
	if (args.length == 1) {
	    return args[0];
	} else {
            return null;
        }
    }

    public static void main(String[] args) {
	final String hello = "Hello";

        String name = getParameter(args);
        if (name == null) {
            name = "World!";
        }

	System.out.println(buildString(hello, name));
    }
}

File: 110.high_level_syntax/src/main/scala/solution/HighLevelSyntax.scala
package solution

object HighLevelSyntax extends App {
  def getParameter(args: Array[String]): Option[String] = {
    if (args.length == 1) {
      Some(args(0))
    } else {
      None
    }
  }

  def buildString(one: String, two: String): String = {
    s"$one $two"
  }

  val hello = "Hello"
  val name = getParameter(args).getOrElse("World!")

  println(buildString(hello, name))
}
